src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -0,0 +1,443 @@
+
+            [TGUI(Label = "GloVe Twitter 50D")]
+            GloVeTwitter50D = 5,
+
+            [TGUI(Label = "GloVe Twitter 100D")]
+            GloVeTwitter100D = 6,
+
+            [TGUI(Label = "GloVe Twitter 200D")]
+            GloVeTwitter200D = 7,
+
+            [TGUI(Label = "fastText Wikipedia 300D")]
+            FastTextWikipedia300D = 8,
+
+            [TGUI(Label = "Sentiment-Specific Word Embedding")]
+            Sswe = 9
+        }
+
+        private static Dictionary<PretrainedModelKind, string> _modelsMetaData = new Dictionary<PretrainedModelKind, string>()
+        {
+             { PretrainedModelKind.GloVe50D, "glove.6B.50d.txt" },
+             { PretrainedModelKind.GloVe100D, "glove.6B.100d.txt" },
+             { PretrainedModelKind.GloVe200D, "glove.6B.200d.txt" },
+             { PretrainedModelKind.GloVe300D, "glove.6B.300d.txt" },
+             { PretrainedModelKind.GloVeTwitter25D, "glove.twitter.27B.25d.txt" },
+             { PretrainedModelKind.GloVeTwitter50D, "glove.twitter.27B.50d.txt" },
+             { PretrainedModelKind.GloVeTwitter100D, "glove.twitter.27B.100d.txt" },
+             { PretrainedModelKind.GloVeTwitter200D, "glove.twitter.27B.200d.txt" },
+             { PretrainedModelKind.FastTextWikipedia300D, "wiki.en.vec" },
+             { PretrainedModelKind.Sswe, "sentiment.emd" }
+        };
+
+        private static Dictionary<PretrainedModelKind, int> _linesToSkipInModels = new Dictionary<PretrainedModelKind, int>()
+            { { PretrainedModelKind.FastTextWikipedia300D, 1 } };
+
+        private string EnsureModelFile(IHostEnvironment env, out int linesToSkip, PretrainedModelKind kind)
+        {
+            linesToSkip = 0;
+            if (_modelsMetaData.ContainsKey(kind))
+            {
+                var modelFileName = _modelsMetaData[kind];
+                if (_linesToSkipInModels.ContainsKey(kind))
+                    linesToSkip = _linesToSkipInModels[kind];
+                using (var ch = Host.Start("Ensuring resources"))
+                {
+                    string dir = kind == PretrainedModelKind.Sswe ? Path.Combine("Text", "Sswe") : "WordVectors";
+                    var url = $"{dir}/{modelFileName}";
+                    var ensureModel = ResourceManagerUtils.Instance.EnsureResource(Host, ch, url, modelFileName, dir, Timeout);
+                    ensureModel.Wait();
+                    var errorResult = ResourceManagerUtils.GetErrorMessage(out var errorMessage, ensureModel.Result);
+                    if (errorResult != null)
+                    {
+                        var directory = Path.GetDirectoryName(errorResult.FileName);
+                        var name = Path.GetFileName(errorResult.FileName);
+                        throw ch.Except($"{errorMessage}\nModel file for Word Embedding transform could not be found! " +
+                            $@"Please copy the model file '{name}' from '{url}' to '{directory}'.");
+                    }
```

Comment:
The user story here doesn't seem *great*. This `ResourceManagerUtils` points people to the problematically named `"[https://aka.ms/tlc-resources/](https://aka.ms/tlc-resources/)"`.

Reply:
problematically named resource has it's own issue #546

Model files which we use (except SSWE which is a 70mb) in range from 160MB(glove50d) to 6 GB(fastText). Which I think impossible to fit into any nuget due to size limitation.

---
In reply to: [205848439](https://github.com/dotnet/machinelearning/pull/545#discussion_r205848439) [](ancestors = 205848439)